webhook_redelivery.py

```diff
@@ -0,0 +1,412 @@
+        help="The amount of seconds to look back for failed deliveries.",
+        required=True,
+    )
+    parser.add_argument(
+        "--github-path",
+        type=str,
+        help=(
+            "The path of the organisation or repository where the webhooks are registered. Should"
+            "be in the format of <organisation> or <organisation>/<repository>."
+        ),
+        required=True,
+    )
+    parser.add_argument(
+        "--webhook-id", type=int, help="The identifier of the webhook.", required=True
+    )
+    args = parser.parse_args()
+
+    github_app_client_id = os.getenv(GITHUB_APP_CLIENT_ID_ENV_NAME)
+    github_app_installation_id = os.getenv(GITHUB_APP_INSTALLATION_ID_ENV_NAME)
+    github_app_private_key = os.getenv(GITHUB_APP_PRIVATE_KEY_ENV_NAME)
+    github_token = os.getenv(GITHUB_TOKEN_ENV_NAME)
+
+    github_auth_details: GithubAuthDetails
+    got_str = (
+        f" Got github_app_client_id = {github_app_client_id},"
+        f" github_app_installation_id = {github_app_installation_id},"
+        f" github_app_private_key = {'***' if github_app_private_key else None},"
+        f" github_token = {'***' if github_app_private_key else None}",
+    )
+    if github_token and (
+        github_app_client_id or github_app_installation_id or github_app_private_key
+    ):
+        raise ArgParseError(
+            "Github auth details are specified in two ways. "
+            "Please specify only one of github token or github app auth details."
+            f"{got_str}"
+        )
+    if github_token:
+        github_auth_details = github_token
+    elif github_app_client_id and github_app_installation_id and github_app_private_key:
+        try:
+            github_auth_details = GithubAppAuthDetails(
+                client_id=github_app_client_id,
+                installation_id=int(github_app_installation_id),
+                private_key=github_app_private_key,
+            )
+        except ValueError as exc:
+            raise ArgParseError(f"Failed to parse github auth details: {exc}") from exc
+    else:
+        raise ArgParseError(
+            "Github auth details are not specified completely. "
+            "Am missing github token or complete set of app auth parameters."
+            f"{got_str}"
+        )
+    webhook_address = WebhookAddress(
+        github_org=args.github_path.split("/")[0],
+        github_repo=args.github_path.split("/")[1] if "/" in args.github_path else None,
+        id=args.webhook_id,
+    )
+
+    return _ParsedArgs(
+        since=args.since, github_auth_details=github_auth_details, webhook_address=webhook_address
+    )
+
+
+# this is checked by integration tests
+def _create_json_output(redelivery_count: int) -> str:  # pragma: no cover
+    """Create a JSON output as strong  with the redelivery count.
+
+    Args:
+        redelivery_count: The number of redelivered webhooks.
+    """
+    return json.dumps({"redelivered": redelivery_count})
```

Comment:
Perhaps this can be in-lined. I know we had a conversation about having functions be specific kinds of functions, we can relax that a bit for one liners like this